cuda/solver/common_trs_kernels.cu

```diff
@@ -0,0 +1,1910 @@
+    for (IndexType i = row_start; i != row_end; i += row_step) {
+        const auto dep = colidxs[i];
+
+        if (is_upper) {
+            // Includes diagonal, entries from the other factor evaluate to
+            // negative
+            if (dep - row <= local_inv_count) {
+                ++row_skip_count;
+            }
+        } else {
+            if (row - dep <= local_inv_count) {
+                ++row_skip_count;
+            }
+        }
+    }
+
+    row_skip_counts[row] = row_skip_count;
+}
+
+
+template <typename ValueType, typename IndexType, bool is_upper>
+__global__ void sptrsvebcrwi_generate_inv_kernel(
+    const IndexType* const rowptrs, const IndexType* const colidxs,
+    const ValueType* const vals, IndexType* const row_skip_counts,
+    ValueType* const band_inv,  // zero initialized
+    uint32* const masks, const size_type n, const bool unit_diag)
+{
+    const auto gid = thread::get_thread_id_flat<IndexType>();
+    const auto inv_block = gid / warp_inverse_size;
+    const auto rhs = gid % warp_inverse_size;
+
+    const auto local_start_row = is_upper ? warp_inverse_size - 1 : 0;
+    const auto local_end_row = is_upper ? -1 : warp_inverse_size;
+    const auto local_step_row = is_upper ? -1 : 1;
+
+#pragma unroll
+    for (IndexType _i = local_start_row; _i != local_end_row;
+         _i += local_step_row) {
+        const auto row = (gid / warp_inverse_size) * warp_inverse_size + _i;
+
+        // Skips entries beyond matrix size, in the last/first block
+        if (row >= n) {
+            continue;
+        }
+
+        // Go though all block-internal dependencies of the row
+
+        const auto row_start = is_upper
+                                   ? rowptrs[row] + row_skip_counts[row] - 1
+                                   : rowptrs[row + 1] - row_skip_counts[row];
+        const auto row_end = is_upper ? rowptrs[row] : rowptrs[row + 1] - 1;
+        const auto row_step = is_upper ? -1 : 1;
+
+        auto sum = zero<ValueType>();
+        IndexType i = row_start;
+        for (; i != row_end; i += row_step) {
+            const auto dep = colidxs[i];
+
+            // To skip out-of-triangle entries for compressed storage
+            if (dep == row) {
+                break;
+            }
+
+            sum +=
+                band_inv[inv_block * (warp_inverse_size * warp_inverse_size) +
+                         dep % warp_inverse_size + rhs * warp_inverse_size] *
+                vals[i];
+        }
+
+        const auto diag = unit_diag ? one<ValueType>() : vals[i];
+        const auto r =
+            ((rhs == _i ? one<ValueType>() : zero<ValueType>()) - sum) / diag;
+
+        band_inv[inv_block * (warp_inverse_size * warp_inverse_size) +
+                 row % warp_inverse_size + rhs * warp_inverse_size] = r;
+    }
+
+
+    if (gid >= n) {
+        return;
+    }
+
+    const auto local_row = rhs;
+    const auto row = gid;
+
+    const auto activemask = __activemask();
```

Comment:
activemask and Volta don't play nice with each other :-( This may only be a subset of all threads in this branch